experiments/TestCase/baroclinic_wave_fvm.jl

```diff
@@ -0,0 +1,346 @@
+
+    # Set up the atmosphere model
+    exp_name = "BaroclinicWave"
+    domain_height::FT = 30e3 # distance between surface and top of atmosphere (m)
+    if with_moisture
+        # hyperdiffusion = EquilMoistBiharmonic(FT(8 * 3600))
+        moisture = EquilMoist{FT}()
+        source = (Gravity(), Coriolis())
+    else
+        # hyperdiffusion = DryBiharmonic(FT(8 * 3600))
+        moisture = DryModel()
+        source = (Gravity(), Coriolis())
+    end
+    model = AtmosModel{FT}(
+        AtmosGCMConfigType,
+        param_set;
+        init_state_prognostic = init_baroclinic_wave!,
+        ref_state = ref_state,
+        turbulence = ConstantKinematicViscosity(FT(0)),
+        # hyperdiffusion = hyperdiffusion,
+        moisture = moisture,
+        source = source,
+    )
+
+    config = ClimateMachine.AtmosGCMConfiguration(
+        exp_name,
+        poly_order,
+        resolution,
+        domain_height,
+        param_set,
+        init_baroclinic_wave!;
+        model = model,
+        numerical_flux_first_order = RoeNumericalFlux(),
+        fv_reconstruction = fv_reconstruction,
+    )
+
+    return config
+end
+
+function main()
+    # add a command line argument to specify whether to use a moist setup
+    # TODO: this will move to the future namelist functionality
+    bw_args = ArgParseSettings(autofix_names = true)
+    add_arg_group!(bw_args, "BaroclinicWave")
+    @add_arg_table! bw_args begin
+        "--with-moisture"
+        help = "use a moist setup"
+        action = :store_const
+        constant = true
+        default = false
+    end
+
+    cl_args = ClimateMachine.init(parse_clargs = true, custom_clargs = bw_args)
+    with_moisture = cl_args["with_moisture"]
+
+    # Driver configuration parameters
+    FT = Float64                             # floating type precision
+    poly_order = (5, 0)                      # discontinuous Galerkin polynomial order
+    fv_reconstruction = FVLinear()         # finite volume reconstruction scheme
+    n_horz = 8                               # horizontal element number
+    n_vert = 20                               # vertical element number
+    # n_days::FT = 1
+    # timestart::FT = 0                        # start time (s)
+    # timeend::FT = n_days * day(param_set)    # end time (s)
+
+    n_seconds::FT = 120
+    timestart::FT = 0                        # start time (s)
+    timeend::FT = n_seconds    # end time (s)
+
+    # Set up driver configuration
+    driver_config = config_baroclinic_wave(
+        FT,
+        poly_order,
+        fv_reconstruction,
+        (n_horz, n_vert),
+        with_moisture,
+    )
+
+    # Set up experiment
+    # ode_solver_type = ClimateMachine.IMEXSolverType(
+    #     implicit_model = AtmosAcousticGravityLinearModel,
+    #     implicit_solver = ManyColumnLU,
+    #     solver_method = ARK2GiraldoKellyConstantinescu,
+    #     split_explicit_implicit = true,
+    #     discrete_splitting = false,
+    # )
```

Comment:
Please check the commented blocks here and below to see if they are stale, please remove them